pkg/reconciler/taskrun/taskrun.go

```diff
@@ -134,6 +137,20 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, tr *v1beta1.TaskRun) pkg
 		// on the event to perform user facing initialisations, such has reset a CI check status
 		afterCondition := tr.Status.GetCondition(apis.ConditionSucceeded)
 		events.Emit(ctx, nil, afterCondition, tr)
+	} else if spireEnabled {
```

Comment:
it's strange that the if-else block has conditions that seem unrelated:
- taskrun has started
- spire is enabled

what happens if both are true, taskrun has not started and spire is enabled, what happens? I'd expect the spire logic to be executed after starting the taskrun 🤔

Reply:
Here we are checking if the previous status matches the signature. 
If the taskrun has not started then this is the first reconcile and we would not have a signature on the status. We need to verify this annotation only on the 2nd reconcile onwards.

Reply:
please add a comment with this explanation so it's clear when someone wants to make changes later

Reply:
Done.